shell.nix

```diff
@@ -0,0 +1,45 @@
+let 
+  pkgs = import <nixpkgs> {};
+  beku = pkgs.callPackage(pkgs.fetchFromGitHub {
+    owner = "stackabletech";
+    repo = "beku.py";
+    rev = "589e48ae45c5984d8f1528e1c0b802e9fa137715";
+    hash = "sha256-hLaIY4BE+VIMeKmS3JLOZy87OC2VuQtbX/NCIbQr2p4=";
+  } + "/beku.nix") {};
+in pkgs.mkShell rec {
+  name = "secret-operator";
+  buildInputs = with pkgs; [
+    # cargo et-al
+    rustup
+
+    # make regenerate charts
+    yq-go
+    git
+    jq
+
+    # make run-dev
```

Comment:
make run-dev should pull in its dependencies via nix already, we don't need them in shell.nix

Reply:
I was getting errors about missing binaries when I ran `make run-dev` without the listed dependencies. I tested in `nix-shell --pure` - does that work for you without the listed deps?

_Btw, you can select multiple consecutive lines to comment on so that there is more context in the comment view._

Reply:
I know.. I was on mobile so I think the selection got screwed up.. :(

Reply:
Ah yeah.

Can this be considered resolved, given my answer? Or have I neglected something?

Reply:
`nix-shell --pure` fails to even evaluate for me:

```
n@emeritus:~/Documents/stackable/secret-provisioner$ nix-shell --pure
building '/nix/store/60prdsmj7p46jjz25dj2zr5iy7fgk377-source.drv'...

trying https://github.com/stackabletech/beku.py/archive/589e48ae45c5984d8f1528e1c0b802e9fa137715.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 22970    0 22970    0     0  44492      0 --:--:-- --:--:-- --:--:-- 44492
unpacking source archive /build/589e48ae45c5984d8f1528e1c0b802e9fa137715.tar.gz
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'secret-operator'
         whose name attribute is located at /nix/store/zyviry1gxp3y8a82myrcvxyr99p2v64m-nixpkgs/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:300:7

       … while evaluating attribute 'buildInputs' of derivation 'secret-operator'

         at /nix/store/zyviry1gxp3y8a82myrcvxyr99p2v64m-nixpkgs/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:347:7:

          346|       depsHostHost                = lib.elemAt (lib.elemAt dependencies 1) 0;
          347|       buildInputs                 = lib.elemAt (lib.elemAt dependencies 1) 1;
             |       ^
          348|       depsTargetTarget            = lib.elemAt (lib.elemAt dependencies 2) 0;

       error: hash mismatch in fixed-output derivation '/nix/store/60prdsmj7p46jjz25dj2zr5iy7fgk377-source.drv':
         specified: sha256-hLaIY4BE+VIMeKmS3JLOZy87OC2VuQtbX/NCIbQr2p4=
            got:    sha256-ueMqyJJuowrLaqrFrs2YpTZVMvwVxU0Xk/T9Cp9pfe4=
```

Reply:
Did you update the revision without updating the hash?

Reply:
Okay, tested after disabling beku.. it looks like I can get away with disabling `tilt` (definitely grabbed from default.nix, at least I'm not entirely crazy...) and `kind`. Crate2nix also failed inside `--pure` unless I also enabled `cacert`. I don't love not pinning down the others, but I don't think that should block this PR.

Reply:
I would expect an impure shell to be used (so using the certs the developer already trusts). I was just using the pure env to catch all the other dependencies that might not already be on the developer machine.

Reply:
@nightkr would you mind trying again when you get the chance?
Some things that have helped others:
- Commit the Cargo.nix and crate-sources.json (which were in .gitignore) 6f472741b850267e659c0b54fa56c85b76317b4e
- Pulling in submodules `git submodule update`

Reply:
The reason that I'm somewhat wary of committing the Cargo.nix is that it's another thing to get out of sync when some people use the Nix workflow and some don't.

Reply:
Do you mean as changes are made by non Nix users, the Nix stuff might lag?
I imagine a sprinkle of automation to keep that up to date might help there (I mean, I can easily forget to update it too as a Nix user).

By hiding it away, it just adds a bit more friction for those who do use nix. They have to keep running an extra step, and then each would be potentially out of date with the other. So I do think it is best that it is centralised/shared.

Is this enough of a show stopper for you, or can we do it and deal with sync issues as they come (I'd rather not preempt too much at this juncture)?

If so, could it be a think we put on the experimental board that was recently discussed by the team?

Reply:
I think it's fine to try out.

> By hiding it away, it just adds a bit more friction for those who do use nix.

The thought process at the time was that it was less about "let's add a Nix workflow" and more "let's add the Tilt workflow and that happens to involve Nix".

Now that that situation is changing, I think it's fair to try checking it in and see how it ends up working out.